modules/thermal_hydraulics/src/components/FileMeshComponent.C

```diff
@@ -0,0 +1,199 @@
+      // - Get the node IDs on the element
+      std::vector<dof_id_type> node_ids(exio_helper.num_nodes_per_elem);
+      for (int k = 0; k < exio_helper.num_nodes_per_elem; k++)
+      {
+        int gi = (j - jmax_last_block) * exio_helper.num_nodes_per_elem + conv.get_node_map(k);
+        int ex_node_id = exio_helper.node_num_map[exio_helper.connect[gi] - 1];
+        node_ids[k] = node_id_map[ex_node_id];
+      }
+
+      // Add the element and set its subdomain
+      Elem * elem = addElement(conv.libmesh_elem_type(), node_ids);
+      elem->subdomain_id() = sid;
+
+      // Populate elem_id_map
+      int exodus_id = exio_helper.elem_num_map[j];
+      elem_id_map[exodus_id] = elem->id();
+    }
+    jmax_last_block += exio_helper.num_elem_this_blk;
+  }
+
+  // Loop over boundaries:
+  // - Get the Exodus boundary names and store them
+  // - Generate the boundary IDs
+  exio_helper.read_sideset_info();
+  int offset = 0;
+  std::unordered_map<BoundaryID, BoundaryName> new_ids_to_names;
+  for (int i = 0; i < exio_helper.num_side_sets; i++)
+  {
+    // Compute new offset
+    offset += (i > 0 ? exio_helper.num_sides_per_set[i - 1] : 0);
+    exio_helper.read_sideset(i, offset);
+
+    // Get boundary name from file (or ID if no name)
+    int ex_sideset_id = exio_helper.get_side_set_id(i);
+    std::string sideset_name = exio_helper.get_side_set_name(i);
+    if (sideset_name.empty())
+      sideset_name = Moose::stringify(ex_sideset_id);
+
+    // Generate the boundary ID for THMMesh and populate boundary_id_map
+    unsigned int bc_id = thm_mesh.getNextBoundaryId();
+    boundary_id_map[ex_sideset_id] = bc_id;
+    new_ids_to_names.emplace(bc_id, sideset_name);
+  }
+
+  auto & boundary_info = thm_mesh.getMesh().get_boundary_info();
+
+  // Loop over the elements on boundaries
+  // Add the boundary elem/side pairs to the boundary
+  for (auto e : index_range(exio_helper.elem_list))
+  {
+    // Get the element
+    int ex_elem_id = exio_helper.elem_num_map[exio_helper.elem_list[e] - 1];
+    dof_id_type elem_id = elem_id_map[ex_elem_id];
+    Elem * elem = thm_mesh.elemPtr(elem_id);
+
+    // Get the side index
+    const auto & conv = exio_helper.get_conversion(elem->type());
+    // Map the zero-based Exodus side numbering to the libmesh side numbering
+    unsigned int raw_side_index = exio_helper.side_list[e] - 1;
+    std::size_t side_index_offset = conv.get_shellface_index_offset();
+    unsigned int side_index = static_cast<unsigned int>(raw_side_index - side_index_offset);
+    int mapped_side = conv.get_side_map(side_index);
+
+    // Get the boundary ID and add the elem/side pair to the boundary
+    unsigned int bc_id = boundary_id_map[exio_helper.id_list[e]];
+    boundary_info.add_side(elem, mapped_side, bc_id);
+  }
+
+  // Generate and set the boundary name
+  for (const auto & id_and_name : new_ids_to_names)
+    boundary_info.sideset_name(id_and_name.first) = genName(_name, id_and_name.second);
+
```

Comment:
the nodeset names are missing. the nodesets are properly passed though

Reply:
I just tried adding this in this loop:
```
boundary_info.nodeset_name(id_and_name.first) = genName(_name, id_and_name.second);
```
but to no effect. The sidesets and nodesets are sharing the IDs, right?

Reply:
they should

Reply:
Yeah, the IDs in Paraview match. It appears to be not just a problem in this component, but other ones as well. And sometimes it's the other way around - the side sets being named as "surface_#". I'll look into this to see if I can see some kind of pattern...

Reply:
Ok, so apparently some gold files have "surface_#", but if I rerun those same tests, I get "nodelist_#" instead. And this appears to happen if we do
```
boundary_info.add_side(elem, side, id);
boundary_info.sideset_name(id) = name;
```
but not when we do
```
boundary_info.add_node(node, id);
boundary_info.nodeset_name(id) = name;
```

I found that this is what fixes it:
```
  for (const auto & id_and_name : new_ids_to_names)
  {
    const std::string sideset_name = genName(_name, id_and_name.second);
    boundary_info.sideset_name(id_and_name.first) = sideset_name;
    boundary_info.nodeset_name(id_and_name.first) = sideset_name;
  }

  boundary_info.build_node_list_from_side_list();
```
It doesn't seem to do anything if I omit the `nodeset_name` line, which surprised me.

`build_node_list_from_side_list()` is already getting executed elsewhere afterwards, so I don't know I need the extra call here, but whatever.